packages/lib/passport-interface/src/SubscriptionManager.ts

```diff
@@ -278,34 +293,66 @@ export class FeedSubscriptionManager {
     const responses: SubscriptionActions[] = [];
     this.resetError(subscription.id);
     try {
-      const authKey = await this.getAuthKeyForFeed(subscription);
+      const authKey = undefined; //await this.getAuthKeyForFeed(subscription);
 
-      const pcdCredential: SerializedPCD | undefined = authKey
+      const pcdCredentials: SerializedPCD[] | undefined = authKey
         ? await this.makeAlternateCredentialPCD(authKey)
-        : await credentialManager.requestCredential({
+        : await credentialManager.requestCredentials({
             signatureType: "sempahore-signature-pcd",
             pcdType: subscription.feed.credentialRequest.pcdType
           });
 
-      const result = await this.api.pollFeed(subscription.providerUrl, {
-        feedId: subscription.feed.id,
-        pcd: pcdCredential
-      });
+      const results = await Promise.all(
+        pcdCredentials.map((credential) => {
+          return this.api.pollFeed(subscription.providerUrl, {
+            feedId: subscription.feed.id,
+            pcd: credential
+          });
+        })
+      );
 
-      if (!result.success) {
-        if (result.code === 410) {
-          this.flagSubscriptionAsEnded(subscription.id, result.error);
+      const failure = results.find((r) => !r.success);
+      if (failure?.success === false) {
+        if (failure.code === 410) {
+          this.flagSubscriptionAsEnded(subscription.id, failure.error);
           return responses;
         }
 
-        throw new Error(result.error);
+        throw new Error(failure.error);
       }
 
-      const { actions } = result.value;
+      const actions = results.flatMap((r) => {
+        if (r.success) {
+          return r.value.actions;
+        }
+        return [];
+      });
+
+      // in the case that we're requesting from the same feed
+      // multiple times using two different email credentials,
+      // the result of one of the requests would overwrite the
+      // result of the other request, given both have a 'delete'
+      // action. thus, we coalesce equivalent delete folder actions
+      // into the the first occurrence of a deletion for each folder.
+      const filteredActions = [];
+      const seen = new Set<string>();
+      for (const action of actions) {
+        if (action.type === "DeleteFolder_action") {
+          const key = action.folder;
+          if (seen.has(key)) {
+            continue;
+          } else {
+            seen.add(key);
+            filteredActions.push(action);
+          }
+        } else {
+          filteredActions.push(action);
+        }
+      }
```

Comment:
another thing to watch out for

Reply:
cancelled this - credential now includes all the user's email PCDs so the feed returns tickets for all their emails in one response, thus we don't need to do this deduplication anymore.